flask_resty/view.py

```diff
@@ -400,9 +400,12 @@ def request_args(self):
                 # KeyError for args that aren't present.
                 continue
 
-            value = args.getlist(field_name)
-            if not isinstance(field, fields.List) and len(value) == 1:
-                value = value[0]
+            if not isinstance(field, fields.List):
```

Comment:
we probably shouldn't support both. perhaps for consistency, just do:

```python
value = args.get(field_name)

if isinstance(field, fields.List):
    # Treat empty strings as empty lists.
    value = value.split(",") if value else []
```

Reply:
cc @sloria

Reply:
the question was more like – for list-like things in args, should we do comma-separated? or should we do multiple values?

or actually should we throw this whole thing out and just use webargs?

Reply:
Ha, probably not worth adding webargs just for this, but..webargs actually both. So webargs will call `args.getlist(...)` when parsing with a `List`. When you need to parse a comma-delimited list, you need to explicitly use [`DelimitedList`](https://github.com/marshmallow-code/webargs/blob/de061e037285fd08a42d73be95bc779f2a4e3c47/src/webargs/fields.py#L47) . We could yoink those bits here. wdyt, @taion ?

Reply:
So if I have a field called `capabilities`, all I have to do is define it as a `DelimitedList` in my args schema to specify that I'll use commas as delimiters? And if I specify it as just `List`, I would use multiple param values?

Reply:
Exactly. So what that would require here is

1. Adding `flask_resty.fields.DelimitedList`
2. Calling `getlist` when parsing with a `List` but not `DelimitedList`

```python
if isinstance(field, ma.fields.List) and not isinstance(DelimitedList):
    value = args.getlist(field_name)

# or duck type, like in webargs
if isinstance(field, ma.fields.List) and not hasattr(field, "delimiter"):
```

Reply:
Can we do even simpler and define our own `DelimitedList` that doesn't extend `fields.List`, _only_ deals with delimited strings, and then just keep the code here the same?

Reply:
Wait, sorry, taking a step back. My question is – if you do `fields.List`, what's a reasonable thing for Flask-RESTy to do?

I don't recall why I decided to have it work with multiple args as opposed to comma-separated lists in the first place. It seems sort of inconsistent with all the `filtering`, &c. stuff.

It's might be weird that e.g. filtering operators only deal with comma-separated values, but it's definitely weird that the `request_args` scheme does something different and incompatible for lists.

Reply:
I think I see what you mean. `filtering` and `sorting` transparently split on commas and pass those to the scalar marshmallow fields, whereas `request_args` expects duplicate args but does not handle comma-separated values. Sorta inconsistent.

I actually think this is fine behavior. `filtering` and `sorting` use a schema that defines the input and output shape for a model. `request_args`/`args_schema` is only dealing with input, so using `List` to handle duplicated query params is reasonable.

Reply:
okay, then i think the way to go is to add a `DelimitedList` field, then... but a slightly different one that only takes strings, i think

Reply:
>  but a slightly different one that only takes strings, i think

To avoid the additional typecheck here? Sure, seems fine to me.

Reply:
That said, I'm not sure there's a way to implement `DelimitedList` as a non-subclass of `List` without duplicating a bunch of `List`'s code. If the goal is to just avoid the typecheck here, it doesn't seem worth it.

Reply:
I suppose you could use composition

```python
class DelimitedList(Field):
    def __init__(self, cls_or_instance, **kwargs):
       super().__init__(**kwargs)
       self._list_field = fields.List(cls_or_instance, **kwargs):

    def _deserialize(self, ....):
         try:
              value = value.split(self.delimiter)
          except AttributeError:
                self.fail('invalid')
          return self._list_field.deserialize(value, ...)
```

But that doesn't feel better than inheritance. wdyt, @taion ?

Reply:
I was just thinking for the sake of simplicity here. Otherwise won't the field get a single-element list with a comma-separated string, and then not be able to deal with it?

Reply:
Not if we add the typecheck. To be clear, I'm proposing:

```python
            value = args.get(field_name)
            if isinstance(field, fields.List) and not hasattr(field, "delimiter"):
                value = args.getlist(field_name)
            data_raw[field_name] = value
```